File: src/app/services/review.service.ts
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { APP_CONFIG } from '../../main';

export interface ReviewDto {
  rating: number;
  comment?: string;
}

@Injectable({ providedIn: 'root' })
export class ReviewService {
  private apiUrl: string;

  constructor(private http: HttpClient, @Inject(APP_CONFIG) private config: any) {
    this.apiUrl = `${this.config.apiUrl}/reviews`;
  }

  getByProduct(productId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/product/${productId}`);
  }

  addReview(productId: number, payload: ReviewDto) {
    return this.http.post<any>(`${this.apiUrl}/product/${productId}`, payload);
  }
}

File: lib/services/review_service.dart
import '../utils/constants.dart';
import 'api_service.dart';

class ReviewService {
  final ApiService _apiService = ApiService();
  final String _baseUrl = '${AppConstants.apiUrl}/reviews';

  Future<List<Map<String, dynamic>>> getByProduct(int productId) async {
    try {
      final resp = await _apiService.get('$_baseUrl/product/$productId');
      if (resp is List) {
        return resp.map((e) => Map<String, dynamic>.from(e as Map)).toList();
      }
      return [];
    } catch (e) {
      return [];
    }
  }

  Future<Map<String, dynamic>> addReview(int productId, Map<String, dynamic> payload) async {
    final resp = await _apiService.post('$_baseUrl/product/$productId', payload);
    return Map<String, dynamic>.from(resp as Map);
  }
}

